From cfed46e137ba2bd69ba51239b4f16e4149123438 Mon Sep 17 00:00:00 2001
From: sidex15 <24408329+sidex15@users.noreply.github.com>
Date: Tue, 8 Jul 2025 11:43:14 +0800
Subject: [PATCH] [BACKPORT] fs: implement SUSFS v1.5.9

There is a few change in here:
- sus_mount functionality still remain in v1.5.5 as backporting it to the latest version will result a mount detection leak in some apps/detectors

Thank you @simonpunk for the help of sus_path backport

Co-authored-by: sidex15 <24408329+sidex15@users.noreply.github.com>
Co-authored-by: simonpunk <simonpunk2016@gmail.com>
---
 fs/Makefile                 |    2 +
 
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/fs/Makefile b/fs/Makefile
index 08fe716e5..fa7ed28b3 100755
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -14,6 +14,8 @@ obj-y :=	open.o read_write.o file_table.o super.o \
 		pnode.o splice.o sync.o utimes.o \
 		stack.o fs_struct.o statfs.o fs_pin.o nsfs.o
 
+obj-$(CONFIG_KSU_SUSFS) += susfs.o
+
 ifeq ($(CONFIG_BLOCK),y)
 obj-y +=	buffer.o block_dev.o direct-io.o mpage.o
 else
